File: Aula_05/python/03.py
class ContaBancaria:
    def __init__ (self):
        self.__nome = ""
        self.__numeroConta = 0
        self.__saldo = 0

    def set_nome (self, v):
        if len(v) != 0:
            self.__nome = v
        else:
            raise ValueError ("Nome vazio")
        
    def set_numero_da_conta (self, v):
        if v > 1:
            self.__numeroConta = v
        else:
            raise ValueError ("Numero da conta nao pode ser um valor abaixo de um")
        
    def get_nome (self):
        return self.__nome
    
    def get_numero_da_conta (self):
        return self.__numeroConta
    
    def get_saldo (self):
        return self.__saldo
    
    def deposito (self, v):
        if v > 1:
            self.__saldo = self.__saldo + v
        else:
            raise ValueError ("Voce so pode depositar valores acima de 1")
        
    def saque (self, v):
        if v < self.__saldo:
            self.__saldo = self.__saldo - v
        else:
            raise ValueError ("Saldo insuficiente")
        
    
class UI:
    @staticmethod
    def main():
        x = ContaBancaria()

        x.set_nome(str(input("Informe seu nome: ")))
        x.set_numero_da_conta(int(input("Informe o numero da sua conta: ")))

        x.deposito(float(input("Informe qual o valor do depoisto: ")))
        x.saque(float(input("Informe o valor do saque: ")))

        print(f"Nome = {x.get_nome()}")
        print(f"Numero da conta = {x.get_numero_da_conta()}")
        print(f"Saldo = {x.get_saldo()}")

UI.main()

File: Aula_05/c++/03.cpp
#include <iostream>
#include <string>

class ContaBancaria {
    private:
    std::string nome;
    int numeroConta;
    double saldo;

    public: 
    void setNome (std::string v){
        if (v.length() > 1) {
            nome = v;
        } else {
            throw std::invalid_argument("Nome vazio");
        }
    }

    void setNumeroDaConta (int v){
        if (v > 1){
            numeroConta = v;
        } else {
            throw std::invalid_argument("Nome da conta nao pode ser um valor negativo");
        }
    }

    std::string getNome () {
        return nome;
    }

    int getNumeroDaConta () {
        return numeroConta;
    }

    double getSaldo (){
        return saldo;
    }

    void deposito (double v) {
        if (v > 1){
            saldo = saldo + v;
        } else {
            throw std::invalid_argument("Voce so pode depositar valores acima de 1");
        }
    }

    void saque (double v){
        if (v < saldo){
            saldo = saldo - v;
        } else {
            throw std::invalid_argument("Saldo insuficiente");
        }
    }
};

class UI{
    public:
    static void main(){
        ContaBancaria x;

        std::string n;
        int v;

        std::cout << "Informe seu nome: ";
        std::cin >> n;
        x.setNome(n);

        std::cout << "Informe o numero da sua conta: ";
        std::cin >> v;
        x.setNumeroDaConta(v);

        float z;
        std::cout << "Informe qual o valor do deposito: ";
        std::cin >> z;
        x.deposito(z);

        std::cout << "Informe o valor do saque: ";
        std::cin >> z;
        x.saque(z);

        std::cout << "Nome = " << x.getNome() << std::endl;
        std::cout << "Numero da conta = " << x.getNumeroDaConta() << std::endl;
        std::cout << "Saldo = " << x.getSaldo() << std::endl;
    }
};

int main(){
    UI::main();
    return 0;
}